### DISCUSSION TITLE
FileTransfer (TemporaryFileTransferType) in milo

### DISCUSSION THREAD

**Discussion by rabi-kumar** [eclipse-milo/milo#1061] [General]
Hi @kevinherron ,

First of all thanks for the great work. 

We are trying to implement the TemporaryFileTransferType (https://reference.opcfoundation.org/v104/Core/docs/Part5/C.4.1/) which I think is from v1.04 and I think milo targets v1.03 specification. 

We have an example server where this kind of node is present with following references:

<img width="762" alt="image" src="https://user-images.githubusercontent.com/42206304/194135241-ceb0db8c-3ae6-48bd-aa12-5a25d216043f.png">

Also, it seems we should be able to transfer the data from server to client, client to server with read to local file and write from local file like below:

<img width="375" alt="image" src="https://user-images.githubusercontent.com/42206304/194135519-69c3d3ea-19c6-4a4c-b2af-651f29a80a9c.png">

I was wondering if it is possible to implement this type with the current milo sdk? If it is possible then I would very grateful to know how this can be achieved. 

Thanks in advance:)

**Comment by kevinherron** [comment]
You would have to create all the necessary classes and behaviors yourself right now.

Work on implementing OPC UA 1.05 is happening right now in the [dev/2.0](https://github.com/eclipse/milo/tree/dev/2.0) branch (ignore the name, it will be version 1.0 when done). I wouldn't recommend using this right now because nothing is stable API-wise, but you can at least see the work happening and what the types you're asking about will look like.
